internal/common/common.go

```diff
@@ -70,3 +71,13 @@ func GetBootstrapHost(cluster *Cluster) *models.Host {
 func IsSingleNodeCluster(cluster *Cluster) bool {
 	return swag.StringValue(cluster.HighAvailabilityMode) == models.ClusterHighAvailabilityModeNone
 }
+
+// IsInPreInstallationState verifies if cluster is in pre-installation stage
+func IsInPreInstallationState(cluster *Cluster) bool {
```

Comment:
Maybe i should be part of cluster API, after all, it's a cluster-specific logic

Reply:
Why was it resovled?

Reply:
i think i missed it after pushing fixes and closing all previous. 
In any case i don't think it should be api call, it will be overkill.

Reply:
I don't think @filanov means it should be an API call just move this function to be under clusterapi package instead of common

Reply:
Yes, there is no reason to keep this logic inside common functions it's a cluster specific function.

Reply:
the problem that this will create circle dependency. Cluster api requires manifests api and then manifests api will require cluster. 
I think this sort of functions doesn't need to be api or needs to be some another common interface.
I still prefer to leave it here.

Reply:
Common is not a garbage collector and the controller should call for inventory internal API.
I don't think that this function belongs to common.

Reply:
Instead of having status all over the code, I think that each component has its own functions and keeps the state in this function, you don't need to import cluster API, you can add another function to `InstallerInternals` that will call to `IsInPreInstallationState` of cluster API.

Reply:
i think we started to over complicate stuff. Really prefer just to remove it and same stuff we did till now by validating list of allowed statuses. As i said what you suggest will cause circle dependency between manifests and clusterApi. I don't want to handle it in this PR for sure

Reply:
IMO it's better than have a spaghetti code